database/sinkdb/sinkdb.go

```diff
@@ -46,6 +50,21 @@ func (db *DB) Ping() error {
 	return err
 }
 
+// Close closes the database handle releasing its resources. It is
+// the caller's responsibility to ensure that there are no concurrent
+// database operations in flight. Close is idempotent.
+//
+// All other methods have undefined behavior on a closed DB.
```

Comment:
We could, but that would only get us part of the way there. The DB could be closed after another method is invoked but before it completes. We'd have to either hold on to the mutex during the call (and allow a single operation at a time) or track the number of inflight operations and in Close wait for all operations to cease before closing. Practically, since this is just intended to make it unit testable, IMO it's better to leave it undefined and make it the caller's responsibility to ensure they don't use a closed DB.